lib/workers/package-file/dep-type.js

```diff
@@ -104,6 +106,9 @@ function getDepConfig(depTypeConfig, dep) {
           }
         }
       }
+      if (applyRule !== false && matchVersion) {
```

Comment:
By applying the check at this late stage it means we only match *if* `applyRule` was already true from a previous match on package name